src/puppetlabs/puppetdb/anonymizer.clj

```diff
@@ -389,8 +389,9 @@
 (defn anonymize-report
   "Anonymize a report"
   [config version report]
-  {:pre  [(report? report version)]
-   :post [(report? % version)]}
+  {:pre  [(report? report)]}
+  {:post [#(report? %)]}
```

Comment:
the :pre and :post should be in the same map, then you can use (report? %). As written this will validate true on output no matter what.